connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -371,6 +371,22 @@ class SparkConnectPlanner(session: SparkSession) {
     }
   }
 
+  private def parseDatatypeString(sqlText: String): DataType = {
+    var dataType: DataType = null
+    try {
+      dataType = session.sessionState.sqlParser.parseTableSchema(sqlText)
+    } catch {
+      case e1: ParseException =>
+        try {
+          dataType = session.sessionState.sqlParser.parseDataType(sqlText)
+        } catch {
+          case e2: ParseException =>
+            dataType = session.sessionState.sqlParser.parseDataType(s"struct<${sqlText.strip}>")
+        }
+    }
+    dataType
```

Comment:
Nit: This can be simplified.
```scala
    try {
      session.sessionState.sqlParser.parseTableSchema(sqlText)
    } catch {
      case _: ParseException =>
        try {
          session.sessionState.sqlParser.parseDataType(sqlText)
        } catch {
          case _: ParseException =>
            session.sessionState.sqlParser.parseDataType(s"struct<${sqlText.strip}>")
        }
    }
```

Reply:
done